libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt

```diff
@@ -73,55 +70,82 @@ class PerformMintUseCase @Inject constructor(
 
         val creator = PublicKey(walletAddress)
 
+        val signer = object : Ed25519Signer() {
+            override val publicKey: ByteArray = creator.pubkey
+            override suspend fun sign(message: ByteArray): ByteArray = ByteArray(signatureLength)
+        }
+
         // create upload files for both metadata and image
         _mintState.value = MintState.CreatingMetadata
 
-        val nftCar = carFileUseCase.buildNftCar(title, desc, filePath)
-
-        val xWeb3Message = web3AuthUseCase
-            .buildxWeb3AuthMessage(creator, nftCar.rootCid.toCanonicalString())
+        val mediaBundle = dataBundleUseCase.buildMediabundle(filePath, signer)
+        val estimatedSize = mediaBundle.byteArray.size +
+                dataBundleUseCase.buildMetadatabundle(title, desc, mediaBundle, signer).byteArray.size
+        val transferTxn = fundNodeRepository.buildNodeFundingTransaction(creator, estimatedSize)
+        transferTxn.setRecentBlockHash(blockhashRepository.getLatestBlockHash())
 
-        // begin message transaction step
-        _mintState.value = MintState.Signing(xWeb3Message.encodeToByteArray())
+        lateinit var metadataBundle: DataItem
 
-        val signatureResult = walletAdapter.transact(sender) {
+        val transferResult = walletAdapter.transact(sender) {
             val reauth = reauthorize(identityUri, iconUri, identityName, authToken)
             persistenceUseCase.persistConnection(
                 reauth.publicKey,
                 reauth.accountLabel ?: "",
                 reauth.authToken
             )
 
-            val signingResult = signMessagesDetached(
-                arrayOf(xWeb3Message.encodeToByteArray()),
-                arrayOf(creator.pubkey)
+            val signer = object : Ed25519Signer() {
```

Comment:
Hmm, I'm never a fan of inline class instantiations. Can this be turned into a class that can be instantiated once and reused?